packages/wallets/src/features/cashier/modules/WithdrawalCrypto/components/WithdrawalCryptoForm/components/WithdrawalCryptoAmountConverter/WithdrawalCryptoAmountConverter.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { Field, FieldProps, useFormikContext } from 'formik';
+import { useActiveWalletAccount, useCurrencyConfig, useExchangeRate } from '@deriv/api';
+import ArrowBold from '../../../../../../../../public/images/ic-back-arrow.svg';
+import { WalletTextField } from '../../../../../../../../components';
+import type { TForm } from '../../WithdrawalCryptoForm';
+import './WithdrawalCryptoAmountConverter.scss';
+
+type TProps = {
+    activeWallet: ReturnType<typeof useActiveWalletAccount>['data'];
+    exchangeRate: ReturnType<typeof useExchangeRate>['data'];
+    getCurrencyConfig: ReturnType<typeof useCurrencyConfig>['getConfig'];
+};
+
+const helperMessageMapper = {
+    decimalPlacesExceeded: (limit: number) => `Up to ${limit} decimal places are allowed.`,
+    insufficientFunds: 'Insufficient funds',
+    invalidInput: 'Should be a valid number.',
+    withdrawalLimitError: (min: string, max: string, currency: string) => {
+        return `The current allowed withdraw amount is ${min} to ${max} ${currency}.`;
+    },
+};
+
+const WithdrawalCryptoAmountConverter = ({ activeWallet, exchangeRate, getCurrencyConfig }: TProps) => {
```

Comment:
Hey man i think you should break down this component into smaller chunks. Maybe can be extracted out to a hook.

Reply:
Hi @thisyahlen-deriv, Yes even I think that should be done. Will try to come-up with something in a time box and if cannot execute it, will keep it for the future 😃.